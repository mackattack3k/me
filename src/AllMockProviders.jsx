import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from './features/theme/defaultTheme';
import GithubMockProvider from './testproviders/GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <GithubMockProvider>{children}</GithubMockProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
