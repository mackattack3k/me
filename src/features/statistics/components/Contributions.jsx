import { useQuery } from '@apollo/client';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import deepGet from '../../deepGet';
import ErrorPage from '../../error/components/ErrorPage';
import ExternalLink from '../../navigation/components/ExternalLink';
import { GITHUB_CONTRIBUTIONS } from '../statisticsQueries';
import StatisticsCard from './StatisticsCard';
import StatisticsCardData from './StatisticsCardData';
import StatisticsComment from './StatisticsComment';

const ContributionsContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  padding: 20px 0;
`;

const NO_HITS = 0;

const Contributions = () => {
  const { t } = useTranslation();
  const { loading, error, data } = useQuery(GITHUB_CONTRIBUTIONS);
  console.log({ loading, error, data });
  if (error) {
    return <ErrorPage />;
  }
  const totalCommitContributions =
    deepGet(data, [
      'user',
      'contributionsCollection',
      'totalCommitContributions'
    ]) || NO_HITS;
  const issueCount = deepGet(data, ['search', 'issueCount']) || 0;
  const latestStar = deepGet(data, [
    'user',
    'starredRepositories',
    'nodes',
    '0'
  ]);
  const { url: latestStarRepoUrl = '', name: latestStarRepoName = '', owner } =
    latestStar || {};
  const { url: latestStarOrgUrl = '', login: latestStarOrg = '' } = owner || {};
  return (
    <ContributionsContainer>
      <StatisticsCard>
        <StatisticsCardData isLoading={loading}>
          {totalCommitContributions}
        </StatisticsCardData>
        <StatisticsComment>{t('git.total_commits')}</StatisticsComment>
      </StatisticsCard>
      <StatisticsCard>
        <StatisticsCardData isLoading={loading}>
          {issueCount}
        </StatisticsCardData>
        <StatisticsComment>{t('git.issue_count')}</StatisticsComment>
      </StatisticsCard>
      <StatisticsCard>
        <StatisticsCardData isLoading={loading}>
          <>
            <ExternalLink to={latestStarOrgUrl}>{latestStarOrg}</ExternalLink>
            <span>/</span>
            <ExternalLink to={latestStarRepoUrl}>
              {latestStarRepoName}
            </ExternalLink>
          </>
        </StatisticsCardData>
        <StatisticsComment>{t('git.latest_star')}</StatisticsComment>
      </StatisticsCard>
    </ContributionsContainer>
  );
};

Contributions.propTypes = {};

export default Contributions;
