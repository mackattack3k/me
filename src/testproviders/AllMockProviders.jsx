import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../features/theme/defaultTheme';
import { ToastProvider } from '../features/toast/components/ToastProvider';
import GithubMockProvider from './GithubMockProvider';

const AllMockProviders = ({ children }) => (
  <ThemeProvider theme={defaultTheme.dark}>
    <ToastProvider>
      <GithubMockProvider>{children}</GithubMockProvider>
    </ToastProvider>
  </ThemeProvider>
);

AllMockProviders.propTypes = {
  children: PropTypes.node.isRequired
};

export default AllMockProviders;
