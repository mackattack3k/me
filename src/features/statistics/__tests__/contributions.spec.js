import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { render, act } from '@testing-library/react';
import wait from 'waait';
import AllMockProviders from '../../../AllMockProviders';
import Contributions from '../components/Contributions';

const mountMockedProvider = async () => {
  const component = render(
    <AllMockProviders>
      <Contributions />
    </AllMockProviders>
  );

  await act(() => wait(0));
  return component;
};
const errorMockProvider = async () => {
  const component = render(
    <AllMockProviders>
      <MockedProvider mocks={undefined}>
        <Contributions />
      </MockedProvider>
    </AllMockProviders>
  );
  await act(() => wait(0));
  return component;
};

describe('contributions', () => {
  it('should render with snapshot', async () => {
    const contributions = await mountMockedProvider();
    expect(contributions).toMatchSnapshot();
  });
  it('should number of commits', async () => {
    const { getByText } = await mountMockedProvider();
    const commits = getByText(/Commits/g);
    expect(commits).toBeDefined();
  });
  it('should show error on data failure', async () => {
    const { getByText } = await errorMockProvider();
    const error = getByText(/Error/g);
    expect(error).toBeDefined();
  });
});
