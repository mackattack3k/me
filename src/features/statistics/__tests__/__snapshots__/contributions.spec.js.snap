// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contributions should render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  position: absolute;
  bottom: 0;
  left: 0;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffca28;
}

.c4:hover {
  color: #c79a00;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
  background-color: #252525;
  min-height: 100px;
  min-width: 200px;
  max-width: 400px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 30px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.05),0 1px 2px rgba(0,0,0,0.1);
  box-sizing: border-box;
  margin: 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 10px 0;
  font-size: 28px;
  color: #ffca28;
  text-shadow: 0 0 10px rgba(0,0,0,0.1);
  word-break: break-word;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.c3 {
  font-size: 14px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px 0;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            0
          </div>
          <div
            class="c3"
          >
            Commits since last year
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            0
          </div>
          <div
            class="c3"
          >
            Issues closed ⛔️
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <a
              class="c4"
              href=""
              rel="noopener noreferrer"
              target="_blank"
            />
            <span>
              /
            </span>
            <a
              class="c4"
              href=""
              rel="noopener noreferrer"
              target="_blank"
            />
          </div>
          <div
            class="c3"
          >
            Is the latest starred repository ⭐️
          </div>
        </div>
      </div>
      <div
        class="c5"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-Axmtr chKkkS"
    >
      <div
        class="sc-AxhCb jUwasP"
      >
        <div
          class="sc-AxgMl fcRPoA"
        >
          0
        </div>
        <div
          class="sc-AxheI fsKHkg"
        >
          Commits since last year
        </div>
      </div>
      <div
        class="sc-AxhCb jUwasP"
      >
        <div
          class="sc-AxgMl fcRPoA"
        >
          0
        </div>
        <div
          class="sc-AxheI fsKHkg"
        >
          Issues closed ⛔️
        </div>
      </div>
      <div
        class="sc-AxhCb jUwasP"
      >
        <div
          class="sc-AxgMl fcRPoA"
        >
          <a
            class="sc-AxiKw eBNHRm"
            href=""
            rel="noopener noreferrer"
            target="_blank"
          />
          <span>
            /
          </span>
          <a
            class="sc-AxiKw eBNHRm"
            href=""
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
        <div
          class="sc-AxheI fsKHkg"
        >
          Is the latest starred repository ⭐️
        </div>
      </div>
    </div>
    <div
      class="sc-AxirZ ejZzxV"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
